src/actions/characterActions.js

```diff
@@ -1,10 +1,31 @@
+import store from "@/store/store"
+
 const actionPrefix = "characterActions/"
 
 export const ADD_CHARACTER = actionPrefix + "ADD_CHARACTER"
+export const EDIT_CHARACTER = actionPrefix + "EDIT_CHARACTER"
 
-export const addCharacter = (character) => {
+export const addCharacter = (newCharacter) => {
   return {
     type: ADD_CHARACTER,
-    newCharacter: character,
+    newCharacter,
+  }
+}
+
+export const editCharacter = (updatedCharacter) => {
+  const state = store.getState()
```

Comment:
You should defiantly not be doing this here. Move this logic to update a character in the reducer